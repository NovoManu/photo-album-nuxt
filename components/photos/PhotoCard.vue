<template>
  <v-flex xs6 sm4 md2>
    <v-dialog v-model="dialog" width="100vh" height="100vh">
      <template v-slot:activator="{ on }">
        <v-card hover ripple tile v-on="on">
          <v-img
            v-if="photo.thumbnailUrl"
            :src="photo.thumbnailUrl"
            height="150px"
          />
          <v-img v-else src="https://via.placeholder.com/150" height="150px" />
          <v-card-title>
            <div>
              <span>{{ photo.title }}</span>
            </div>
          </v-card-title>
        </v-card>
      </template>
      <v-card hover ripple tile>
        <v-img
          :src="photo.url"
          width="auto"
          height="auto"
          max-height="100%"
          max-width="100%"
          contain
          class="grey darken-4"
        />
      </v-card>
    </v-dialog>
  </v-flex>
</template>
<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { Photo } from '~/utils/types'

@Component
export default class PhotoCard extends Vue {
  @Prop({ required: true }) photo!: Photo
  dialog: boolean = false
}
</script>
