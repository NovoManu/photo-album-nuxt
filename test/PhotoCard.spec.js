import { mount } from '@vue/test-utils'
import PhotoCard from '@/components/photos/PhotoCard.vue'

describe('index.vue', () => {
  const wrapper = mount(PhotoCard, {
    propsData: {
      photo: {},
      albumTitle: 'title',
      userName: 'name'
    }
  })
  it('renders component', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has initial colors', () => {
    expect(wrapper.vm.dialog).toBe(false)
  })
  it('has initial visible', () => {
    expect(wrapper.vm.visible).toBe(false)
  })
})
