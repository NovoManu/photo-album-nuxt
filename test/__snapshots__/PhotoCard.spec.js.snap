// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue renders component 1`] = `
<div
  class="card d-flex"
>
  <v-layout
    align-start=""
    row=""
    wrap=""
  >
    <v-dialog
      max-width="600"
      width="100%"
    >
       
      <!---->
    </v-dialog>
     
    <!---->
  </v-layout>
</div>
`;
